### PR TITLE
Fix waterfall remote setup for v8 and waterfall

### PR DESCRIPTION
**Pull request by dschuff**
v8 and the waterfall repo itself are special snowflakes.

### CODE REVIEW

File: src/build.py [hunk 1]
```diff
@@ -388,9 +389,13 @@ def pretty(fmt):
       return proc.check_output(
           ['git', 'log', '-n1', '--pretty=format:%s' % fmt],
           cwd=self.src_dir).strip()
-    remote_name = 'remote.%s.url' % WATERFALL_REMOTE
-    remote = proc.check_output(['git', 'config', '--get', remote_name],
-                               cwd=self.src_dir).strip()
+    try:
+      remote = GitRemoteUrl(self.src_dir, WATERFALL_REMOTE)
+    except proc.CalledProcessError:
+      # Not all checkouts have the '_waterfall' remote (e.g. the waterfall
```
Comment (jgravelle-google) [inline src/build.py]: The waterfall syncs itself? I thought its custom_sync was set to NoSync meaning it didn't sync?

File: src/build.py [hunk 2]
```diff
@@ -332,6 +332,17 @@ def RemoteBranch(branch):
   return WATERFALL_REMOTE + '/' + branch
 
 
+def GitUpdateRemote(src_dir, git_repo, remote_name):
```
Comment (jgravelle-google) [inline src/build.py]: This really does belong more as a top-level construct. I had felt it odd to have a function that took three arguments where two of which were member variables when it was only used inside that class.
So, nice!

**Comment by jgravelle-google** [comment]
LGTM

File: src/build.py (hunk 1 above)
Comment (dschuff) [inline src/build.py]: It doesn't sync itself, so there won't be a `_waterfall` remote, but this code (to print the git info) runs on both the waterfall and its sub-checkouts.

**Pull request merged**